crates/lib/src/compiler/libquil.rs

```diff
@@ -0,0 +1,325 @@
+            .symbols
+            .into_iter()
+            .map(CString::new)
+            .collect::<Result<Vec<_>, _>>()?;
+        let result = libquil_sys::quilc::conjugate_pauli_by_clifford(
+            request
+                .pauli
+                .indices
+                .into_iter()
+                .map(u32::try_from)
+                .collect::<Result<_, _>>()?,
+            pauli_terms,
+            &request.clifford.parse()?,
+        )?;
+        Ok(quilc::ConjugatePauliByCliffordResponse {
+            phase: i64::from(result.phase),
+            pauli: result.pauli,
+        })
+    }
+
+    fn generate_randomized_benchmarking_sequence(
+        &self,
+        request: quilc::RandomizedBenchmarkingRequest,
+    ) -> Result<quilc::GenerateRandomizedBenchmarkingSequenceResponse, Self::Error> {
+        let gateset = request
+            .gateset
+            .iter()
+            .map(String::as_str)
+            .map(str::parse)
+            .collect::<Result<Vec<_>, _>>()?;
+        let interleaver = request
+            .interleaver
+            .map(|s| s.parse::<libquil_sys::quilc::Program>())
+            .transpose()?;
+        let result = libquil_sys::quilc::generate_rb_sequence(
+            i32::try_from(request.depth)?,
+            i32::try_from(request.qubits)?,
+            gateset.iter().collect(),
+            request.seed.map(i32::try_from).transpose()?,
+            interleaver.as_ref(),
+        )?;
+        Ok(quilc::GenerateRandomizedBenchmarkingSequenceResponse {
+            sequence: result
+                .into_iter()
+                .map(|i| i.into_iter().map(Into::into).collect())
+                .collect(),
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::{
+        compiler::quilc::{
+            Client as _, CompilerOpts, ConjugateByCliffordRequest,
+            ConjugatePauliByCliffordResponse, GenerateRandomizedBenchmarkingSequenceResponse,
+            PauliTerm, RandomizedBenchmarkingRequest, TargetDevice,
+        },
+        qvm::{self, http::AddressRequest},
+    };
+
+    use super::*;
+
+    use qcs_api_client_openapi::models::InstructionSetArchitecture;
+    use quil_rs::quil::Quil;
+    use regex::Regex;
+    use std::{collections::HashMap, fs::File, num::NonZeroU16};
+
+    const EXPECTED_H0_OUTPUT: &str = "MEASURE 0\n";
+
+    fn aspen_9_isa() -> InstructionSetArchitecture {
+        serde_json::from_reader(File::open("tests/aspen_9_isa.json").unwrap()).unwrap()
+    }
+
+    pub(crate) fn qvm_isa() -> InstructionSetArchitecture {
+        serde_json::from_reader(File::open("tests/qvm_isa.json").unwrap()).unwrap()
+    }
+
+    fn quilc_client() -> Client {
+        Client {}
+    }
+
+    #[tokio::test]
+    async fn compare_native_quil_to_expected_output() {
+        let output = quilc_client()
+            .compile_program(
+                "MEASURE 0",
+                TargetDevice::try_from(qvm_isa()).expect("Couldn't build target device from ISA"),
```

Comment:
`expect` messages should indicate [what is expected](https://doc.rust-lang.org/stable/std/result/enum.Result.html#recommended-message-style) to happen (also below)

Reply:
Agreed. This was copy/pasted from a different file with the same issue.

Reply:
I'll fix this when I improve the testing for libquil.